src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/base/Functions.java

```diff
@@ -292,6 +293,22 @@ public static RuntimeIterator getBuiltInFunctionIterator(
         }
     }
 
+    public static RuntimeIterator getUserDefinedFunctionIterator(
+            FunctionIdentifier identifier,
+            IteratorMetadata metadata,
+            List<RuntimeIterator> arguments
+    ) {
+        return buildUserDefinedFunctionIterator(getUserDefinedFunction(identifier, metadata), metadata, arguments);
```

Comment:
I think getUserDefinedFunction(identifier, metadata) can be simplified to not have the metadata parameter. It seems unused

Reply:
Also good catch, done.